File: 1572-matrix-diagonal-sum/1572-matrix-diagonal-sum.py
class Solution:
    def diagonalSum(self, mat: List[List[int]]) -> int:
        diagonal = [mat[i][i] for i in range(len(mat))]
        sec_diagonal = [mat[i][len(mat) - 1 - i] for i in range(len(mat))]
        if len(mat) % 2 == 1:
            return sum(diagonal) + sum(sec_diagonal) - mat[len(mat)//2][len(mat)//2]
        else:
            return sum(diagonal) + sum(sec_diagonal)

File: 1572-matrix-diagonal-sum/1572-matrix-diagonal-sum.js
/**
 * @param {number[][]} mat
 * @return {number}
 */
var diagonalSum = function (mat) {
  const n = mat.length;
  let answer = 0;
  for (var i = 0; i < n; i++) {
    answer += mat[i][i];
    if (i != n - i - 1) {
      answer += mat[i][n - i - 1];
    }
  }
  return answer;
};
